### DISCUSSION TITLE
Implement a rotation key for dagster-ssh

### DISCUSSION THREAD

**Discussion by slamer59** [dagster-io/dagster#9214] [Q&A] [answered]
Hello, 
I want to implement a rotation key for dagster-ssh. 

I am using the **dagster-ssh** with a [key_string](https://github.com/dagster-io/dagster/blob/b33565cc0881f3363612a3e2a76eb2cbe2986826/python_modules/libraries/dagster-ssh/dagster_ssh/resources.py#L39)

This key will rotate so I need to make a call to some secretmanager **before** using dagster-ssh to be sure that the key is updated. 

How would you do that ? 

Is it possible to have multiple decorators ? 

The first decorator will get the rotated key. 

```
@get_rotated_key
@op(@op(required_resource_keys={"sftp_server_resource"})
def my_ops(context):
    ...

```

**Comment by OwenKephart** [comment, marked as answer]
Hi @slamer59 ! I think the most straightforward solution here would be to create your own custom resource, which is a subclass of the SSHResource. You could put your custom secret manager logic inside the constructor of this new class, to make sure you have an up-to-date key string (rather than hard coding that key string in configuration).

**Reply by slamer59** [reply]
Well, it seems easier and clearer :) 

Just for the logic, I wonder if a new resource is created everytime or if there is a new instance everytime.

**Reply by OwenKephart** [reply]
In almost all cases (basically, unless you're explicitly using the in_process_executor), a new resource instance will be created for each step
